### DISCUSSION TITLE
What's wrong with barrel files?

### DISCUSSION THREAD

**Discussion by AjaxSolutions** [pmndrs/jotai#2291] [General] [answered]
Devs in this [thread](https://twitter.com/ryanflorence/status/1731697953780154869) seem to think that they should be avoided.    Use deep imports instead.

Barrel files are used extensively by Jotai,

**Comment by DavidJFelix** [comment]
I feel like this is largely a matter of opinion and not specific to jotai

**Comment by AjaxSolutions** [comment]
You shouldn't close this discussion so quickly.  There are some objective facts about barrels which disagree with your comment:

"As I heard barrels make tree shaking harder. Maybe I should switch to deep imports."

**Comment by dai-shi** [comment, marked as answer]
We are using it for bundling. It's a different perspective from app development.
See the bundled output: https://unpkg.com/browse/jotai@2.6.0/esm/ (ignore *.mts files, which can't be bundled together yet, unfortunately.)
